src/game_api/rpc.cpp

```diff
@@ -423,7 +423,7 @@ std::vector<uint32_t> get_entities_at(ENT_TYPE entity_type, uint32_t mask, float
             {
                 auto [ix, iy] = item->position();
                 float distance = sqrt(pow(x - ix, 2.0f) + pow(y - iy, 2.0f));
-                if (((item->type->search_flags & mask) > 0 || mask == 0) && (item->type->id == entity_type || entity_type == 0) && distance < radius)
+                if (((item->type->search_flags & mask) > 0 || mask == 0) && distance < radius && (entity_types[0] == 0 || std::find(entity_types.begin(), entity_types.end(), item->type->id) != entity_types.end()))

```

Comment:
This sort of API should probably handle receiving empty `entity_types` the same way it handles `{0}`. So we can add a utility to check for a match and defer all these functions to that.